pyomo/contrib/fme/fourier_motzkin_elimination.py

```diff
@@ -322,3 +370,83 @@ def _add_linear_constraints(self, cons1, cons2):
+        solver_factory: A SolverFactory object (constructed with a solver 
+                        which can solve the continuous relaxation of the 
+                        active constraints on the model. That is, if you 
+                        had nonlinear constraints unrelated to the variables 
+                        being projected, you need to either deactivate them or 
+                        provide a solver which will do the right thing.)
+        """
+        # make sure m looks like what we expect
+        if not hasattr(m, "_pyomo_contrib_fme_transformation"):
+            raise RuntimeError("It looks like model %s has not been "
+                               "transformed with the "
+                               "fourier_motzkin_elimination transformation!"
+                               % m.name)
+        transBlock = m._pyomo_contrib_fme_transformation
+        constraints = transBlock.projected_constraints
+    
+        # relax integrality so that we can do this with LP solves.
+        TransformationFactory('core.relax_integer_vars').apply_to(
+            m, transform_deactivated_blocks=True)
+        # deactivate any active objectives on the model, and save what we did so
+        # we can undo it after.
+        active_objs = []
+        for obj in m.component_data_objects(Objective, descend_into=True):
+            if obj.active:
+                active_objs.append(obj)
+            obj.deactivate()
+        # add placeholder for our own objective
+        obj_name = unique_component_name(m, '_fme_post_process_obj')
+        obj = Objective(expr=0)
+        m.add_component(obj_name, obj)
+        for i in constraints:
+            # If someone wants us to ignore it and leave it in the model, we
+            # can.
+            if not constraints[i].active:
+                continue
+            # deactivate the constraint
+            constraints[i].deactivate()
+            m.del_component(obj)
+            # make objective to maximize its infeasibility
+            obj = Objective(expr=constraints[i].body - constraints[i].lower)
+            m.add_component(obj_name, obj)
+            results = solver_factory.solve(m)
+            print(results.solver.termination_condition)
+            if results.solver.termination_condition == \
+               TerminationCondition.unbounded:
+                obj_val = -float('inf')
+            elif results.solver.termination_condition != \
+               TerminationCondition.optimal:
+                raise RuntimeError("Unsuccessful subproblem solve when checking"
+                                   "constraint %s.\n\t"
+                                   "Termination Condition: %s" %
+                                   (constraints[i].name, 
+                                    results.solver.termination_condition))
+            else:
+                obj_val = value(obj)
+            # if we couldn't make it infeasible, it's useless
+            if obj_val >= 0:
```

Comment:
Should this have a tolerance?  If there is a feasible solution where it is active, then it could end up with a non-zero dual and be relevant to a downstream algorithm?

Reply:
I'm not sure I understand... If this is 0, then it could never be violated. So yes, it could be active, but by keeping this around, we're making that solution degenerate. Would we want that?